third_party/move/move-compiler-v2/tests/cyclic-instantiation-checker/v1-tests/complex_1.exp

```diff
@@ -0,0 +1,29 @@
+
+Diagnostics:
+error: cyclic type instantiation: a cycle of recursive calls causes a type to grow without bound
+   ┌─ tests/cyclic-instantiation-checker/v1-tests/complex_1.move:13:9
+   │
+13 │     fun c<T1, T2>() {
+   │         ^
+   │
+   = `c<T1, T2>` calls `d<T2>` at tests/cyclic-instantiation-checker/v1-tests/complex_1.move:13
+   = `d<T2>` calls `b<u64, T2>` at tests/cyclic-instantiation-checker/v1-tests/complex_1.move:15
+   = `b<u64, T2>` calls `c<M::S<T2>, bool>` at tests/cyclic-instantiation-checker/v1-tests/complex_1.move:20
```

Comment:
I hate to bring bad news, but now that the error messages are very clear I can see that this case doesn't cause a growing type:
```
c<T1, T2> -> d<T2> -> b<u64, T2> -> c<M::S<T2>, bool> -> d<bool> -> b<u64, bool> -> c<M::S<bool>, bool> -> ...
```
If it's not trivial to fix, let's file an issue to revisit later, since this is pretty unlikely.

* Either fix or file an issue and point to this test output.  Either way is fine.
*